src/core/viz/parser.js

```diff
@@ -179,3 +179,78 @@ function cleanJsep() {
     jsep.addLiteral('true');
     jsep.addLiteral('false');
 }
+
+/**
+ * Remove comments from string
+ * - // line comments
+ * - /* block comments
+ * - ' " Keep comments inside single and double quotes
+ * Based on: https://j11y.io/javascript/removing-comments-in-javascript/
+ */
+export function cleanComments(str) {
+    var mode = {
+        singleQuote: false,
+        doubleQuote: false,
+        blockComment: false,
+        lineComment: false
+    };
```

Comment:
🤔 So this is basically a state machine but you keep track of the state with a boolean per state...  so you do `if (mode.singleQuote)` instead of `mode == SINGLEQUOTE`. Am I missing anything?